rmc/utils/constraint.py

```diff
@@ -1890,22 +1975,33 @@ def add_globals_rmc_browser(ht: hl.Table) -> hl.Table:
         - outlier transcripts
     :param HT: Input Table. Should be RMC regions HT annotated with amino acid
         information for region starts and stops.
+    :param filter_to_canonical: Whether to filter to canonical transcripts only.
+    :param keep_outliers: Whether to keep outlier transcripts.
+        Default is True.
     :return: RMC regions HT with updated globals annotations.
     """
     # Get transcripts with evidence of RMC
     rmc_transcripts = hl.literal(ht.aggregate(hl.agg.collect_as_set(ht.transcript)))
 
-    # Get all QC pass transcripts and outlier transcripts
-    qc_pass_transcripts = get_constraint_transcripts(outlier=False)
-    outlier_transcripts = get_constraint_transcripts(outlier=True)
+    # Get all QC pass transcripts
+    qc_pass_transcripts = get_constraint_transcripts(
+        filter_to_canonical=filter_to_canonical, outlier=False
+    )
+    outlier_transcripts = get_constraint_transcripts(
+        filter_to_canonical=filter_to_canonical, outlier=True
+    )
 
-    # Get all transcripts displayed on browser
-    transcript_ht = gene_model.ht()
-    all_transcripts = transcript_ht.aggregate(
-        hl.agg.collect_as_set(transcript_ht.transcript)
+    # Get all transcripts from constraint HT
+    all_transcripts = get_constraint_transcripts(
+        all_transcripts=True, filter_to_canonical=filter_to_canonical
     )
-    transcripts_no_rmc = qc_pass_transcripts.difference(rmc_transcripts)
-    transcripts_not_searched = all_transcripts.difference(qc_pass_transcripts)
+    if keep_outliers:
+        transcripts_no_rmc = all_transcripts.difference(rmc_transcripts)
+        transcripts_not_searched = hl.empty_array(hl.tstr)
+    else:
+        transcripts_no_rmc = qc_pass_transcripts.difference(rmc_transcripts)
+        transcripts_not_searched = all_transcripts.difference(qc_pass_transcripts)
```

Comment:
This should be just `outlier_transcripts` right? I'm ok with keeping it this way though

Reply:
oh yeah, that's true. OK, will leave 😆